control-plane-operator/controllers/hostedcontrolplane/v2/dnsoperator/component.go

```diff
@@ -0,0 +1,53 @@
+package dnsoperator
+
+import (
+	hyperv1 "github.com/openshift/hypershift/api/hypershift/v1beta1"
+	oapiv2 "github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/v2/oapi"
+	component "github.com/openshift/hypershift/support/controlplane-component"
+	"github.com/openshift/hypershift/support/util"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+const (
+	ComponentName = "dns-operator"
+)
+
+var _ component.ComponentOptions = &dnsOperator{}
+
+type dnsOperator struct {
+}
+
+// IsRequestServing implements controlplanecomponent.ComponentOptions.
+func (d *dnsOperator) IsRequestServing() bool {
+	return false
+}
+
+// MultiZoneSpread implements controlplanecomponent.ComponentOptions.
+func (d *dnsOperator) MultiZoneSpread() bool {
+	return true
+}
+
+// NeedsManagementKASAccess implements controlplanecomponent.ComponentOptions.
+func (d *dnsOperator) NeedsManagementKASAccess() bool {
+	return false
+}
+
+func NewComponent() component.ControlPlaneComponent {
+	return component.NewDeploymentComponent(ComponentName, &dnsOperator{}).
+		WithAdaptFunction(adaptDeployment).
+		WithManifestAdapter(
+			"kubeconfig.yaml",
+			component.WithAdaptFunction(adaptKubeconfigSecret),
+			component.DisableIfAnnotationExist(hyperv1.DisablePKIReconciliationAnnotation),
+		).
+		WithDependencies(oapiv2.ComponentName).
+		InjectAvailabilityProberContainer(util.AvailabilityProberOpts{
+			KubeconfigVolumeName: "kubeconfig",
```

Comment:
looking at the ControlPlaneComponent resource for the dns-operator
```
 - lastTransitionTime: "2025-02-10T14:46:07Z"
    message: 'Deployment.apps "dns-operator" is invalid: spec.template.spec.initContainers[0].volumeMounts[0].name:
      Not found: "kubeconfig"'
    reason: ReconciliationError
    status: "False"
    type: Progressing
```
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_hypershift/5593/pull-ci-openshift-hypershift-main-e2e-aws/1888948796826587136/artifacts/e2e-aws/hypershift-aws-run-e2e-external/artifacts/TestCreateClusterV2/namespaces/e2e-clusters-qpdj9-example-drk68/hypershift.openshift.io/controlplanecomponents/dns-operator.yaml

this should be `dns-operator-kubeconfig` instead.